docs/rfcs/003-consolidate-core-packages.md

```diff
@@ -0,0 +1,338 @@
+
+1. Clone the `tree-view` repo
+2. Run `apm publish` to publish a new release of the package
+3. Edit `package.json` in the Atom repo to reflect the new version of `tree-view`
+4. Commit and push the changes to the relevant branch where the change belongs (`master` or `1.nn-releases`)
+
+#### Simplified Package Development
+
+If we were to move `tree-view` (or any other core Atom package) back into `atom/atom`, the development workflow would look more like this:
+
+1. Fork and clone `atom/atom` and switch to a release branch if necessary
+1. Build Atom and launch it in dev mode
+1. Make desired changes to the code in `packages/tree-view`
+1. Open a PR on `atom/atom` and wait for CI to pass and a maintainer to review it
+1. Work with maintainers to get the PR approved and merged
+
+At this point, the change is merged into Atom and ready for inclusion in the next release.
+
+### Greatly reduced burden for maintainers
+
+Since packages all have their own repositories, this means that we have to watch 91 different repos for issues and pull requests.  This also means that we have to redirect issues filed on `atom/atom` to the appropriate repository when a user doesn't know where it belongs.  Even more importantly, there's not an easy way to prioritize and track issues across the Atom organization without using GitHub Projects.
+
+Also, as mentioned above, there's the added duty of doing the package "version dance" when we merge any new PRs to a package repository: publish the package update, update `package.json` in Atom.  It's very easy to forget to do this and not have community contributions included in the next Atom release!
+
+The more core packages live in `atom/atom`, the less work Atom maintainers have to do overall.
+
+## Explanation
+
+Many of Atom's core packages now live in the core `atom/atom` repository.  To the Atom user, this change will be imperceptible as these packages still show up in the list of Core Packages in the Settings View.  Users can still disable these packages if they want to replace their behavior with other packages.
+
+For maintainers and contributors, there will be less juggling of repositories and no more publishing of updates to these packages with `apm`:
+
+Contributors now clone and build `atom/atom` to work on improvements to core packages.  They will no longer have to use `apm link` in dev mode to test changes they make to packages in the repo's `packages` folder.  Core packages that aren't consolidated still have folders under `packages` with README.md files that point to the home repository for that package.
+
+When a contributor sends a PR to `atom/atom` that only affects files in a folder under `packages`, only the specs for the relevant package folders will be executed using Atom's CI scripts.  This means that a full Atom build will not be required when no Atom Core code is changed in a PR.  Package specs are also now run against all 3 OSes on Atom `master` and release builds.
```

Comment:
👍 I like that we get faster feedback on PR builds, but still have a full build cycle to validate the changes in context. (And our full build times could likely be dropped further with another round of optimizations at some point as well...)